src/compiler/checker.ts

```diff
@@ -26804,11 +26819,11 @@ namespace ts {
             }
             const prop = getPropertyOfType(type, propertyName);
             if (prop) {
-                if (isPropertyAccessExpression(node) && prop.valueDeclaration && isPrivateIdentifierPropertyDeclaration(prop.valueDeclaration)) {
+                if (prop.valueDeclaration && isPrivateIdentifierClassElementDeclaration(prop.valueDeclaration)) {
                     const declClass = getContainingClass(prop.valueDeclaration);
                     return !isOptionalChain(node) && !!findAncestor(node, parent => parent === declClass);
                 }
-                return checkPropertyAccessibility(node, isSuper, type, prop);
+                return checkPropertyAccessibility(node, isSuper, type, prop, /* reportError */ false);

```

Comment:
So what exactly are we trying to suppress here? It's unclear why this is needed.

Reply:
I added a check in `getSuggestedSymbolForNonexistentProperty` so that it only suggests accessible members. To do this is use `isValidPropertyAccessForCompletions` which in turn uses `checkPropertyAccessibility`. 

```ts
props = filter(props, prop => isValidPropertyAccessForCompletions(parent, containingType, prop));
```
The problem is that since `checkPropertyAccessibility` reports errors, when the compiler tries to get the suggestion, it will report access errors on all tested properties in that `filter` call. Suppressing the error seems to cause no other issues, `isValidPropertyAccessForCompletions`. 

I could remove this improvement and allow invalid suggestions but I think this is a net improvement.